### PR TITLE
add a test that reduces the size of the producer set

### PR DESCRIPTION
**Pull request by wanderingbort**
**Change Description**

Add an additional test to the producer_schedule_tests that decreases the size of the producer set from 3 to 2 

Add an additional test to the producer_schedule_tests that verifies that a proposed empty producer schedule cannot become active and will not cause gaps to form in the schedule versions of active producer schedules.

**Consensus Changes**

**API Changes**

**Documentation Additions**

### CODE REVIEW

File: unittests/producer_schedule_tests.cpp [hunk 1]
```diff
@@ -260,4 +260,61 @@ BOOST_FIXTURE_TEST_CASE( producer_schedule_promotion_test, TESTER ) try {
...
+      return std::equal( a.begin(), a.end(), b.producers.begin(), b.producers.end() );
+   };
+
+   auto res = set_producers( {N(alice),N(bob),N(carol)} );
+   vector<producer_key> sch1 = {
+                                 {N(alice), get_public_key(N(alice), "active")},
+                                 {N(bob),   get_public_key(N(bob),   "active")},
+                                 {N(carol),   get_public_key(N(carol),   "active")}
+                               };
+   //wlog("set producer schedule to [alice,bob]");
```
Comment (heifner) [inline unittests/producer_schedule_tests.cpp]: remove

File: unittests/producer_schedule_tests.cpp [hunk 2]
```diff
@@ -260,4 +260,61 @@ BOOST_FIXTURE_TEST_CASE( producer_schedule_promotion_test, TESTER ) try {
...
+   BOOST_CHECK_EQUAL( control->active_producers().version, 1 );
+   BOOST_CHECK_EQUAL( true, compare_schedules( sch1, control->active_producers() ) );
+   produce_blocks(7);
+
+   res = set_producers( {N(alice),N(bob)} );
+   vector<producer_key> sch2 = {
+                                 {N(alice), get_public_key(N(alice), "active")},
+                                 {N(bob),   get_public_key(N(bob),   "active")}
+                               };
+   wlog("set producer schedule to [alice,bob,carol]");
```
Comment (heifner) [inline unittests/producer_schedule_tests.cpp]: This should be [alice,bob], right?
